pkg/alertmanager/validation.go

```diff
@@ -81,11 +81,77 @@ func validateReceivers(receivers []monitoringv1alpha1.Receiver) (map[string]stru
 		if err := validatePushoverConfigs(receiver.PushoverConfigs); err != nil {
 			return nil, errors.Wrapf(err, "failed to validate 'pushOverConfig' - receiver %s", receiver.Name)
 		}
+
+		if err := validateSNSConfigs(receiver.SNSConfigs); err != nil {
+			return nil, errors.Wrapf(err, "failed to validate 'snsConfig' - receiver %s", receiver.Name)
+		}
 	}
 
 	return receiverNames, nil
 }
 
+func validateSNSConfigs(configs []monitoringv1alpha1.SNSConfig) error {
+	for _, config := range configs {
+		if config.PhoneNumber == "" && config.TargetARN == "" && config.TopicARN == "" {
+			return errors.New("either 'phone_number', 'target_arn' or 'topic_arn' must be set")
+		}
+
+		if config.PhoneNumber != "" {
+			if len(config.PhoneNumber) > 16 || config.PhoneNumber[0] != '+' {
+				return errors.New("'phone_number' must start with a plus sign followed by a maximum of 15 digits")
+			}
+			if _, err := strconv.Atoi(config.PhoneNumber[1:]); err != nil {
+				return errors.New("'phone_number' must start with a plus sign followed by a maximum of 15 digits")
+			}
```

Comment:
IIUC the checks aren't present in Alertmanager. Also since the phone number string can be a template, they may prevent valid configurations.

Reply:
yes, you are correct, sorry I missed that.